src/software/amazon/ion/impl/IonWriterUser.java

```diff
@@ -374,7 +372,7 @@ public void stepIn(IonType containerType) throws IOException
         // see if it looks like we're starting a local symbol table
         if (containerType == IonType.STRUCT
             && _current_writer.getDepth() == 0
-            && has_annotation(ION_SYMBOL_TABLE, ION_SYMBOL_TABLE_SID))
+            && findAnnotation(ION_SYMBOL_TABLE) == 0)
```

Comment:
If the value in question has a lot of annotations, `findAnnotations` will potentially visit all of them instead of just looking at the first one. Probably not important since most values have 0 or 1 annotation, but I wanted to mention it.